src/components/portfolio/EditDialogAddStock.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import {
+  CircularProgress,
+  TableCell,
+  TableRow,
+  TextField,
+} from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { useTranslation } from 'react-i18next';
+import { stockSearch, StockSearchResult } from '../../portfolio/APIClient';
+import { AddEntryProps } from './EditDialog';
+import StyledNumberFormat from '../shared/StyledNumberFormat';
+
+/** A search bar that adds stocks to an edit dialog. */
```

Comment:
The type `AddEntryProps` with `ids` and `add` is defined and documented in EditDialog.tsx. That's because `EditDialog` is intended as a generic component that allows editing lists of numbers. That component doesn't use the concepts of stocks or portfolios so it's reusable. It only specifies an "interface" for components that add entries to the list. `EditDialogAddStock` is an implementation of that "interface". It's the only implementation, but there could be multiple.

An additional documentation here would be redundant. A decent IDE is able to show the types and docstrings of the parameters, so I think it's not necessary to copy the docstrings from EditDialog.tsx.

Reply:
OK, that's fair, I would have added them just for consistency purposes but if it's just a copy-paste of an existing doc then there's no reason to do that